src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/RuntimeBroker.cs

```diff
@@ -0,0 +1,399 @@
+        /// <param name="allProperties"></param>
+        /// <returns></returns>
+        public MsalTokenResponse ParseWamResponse(
+                NativeInterop.AuthResult authResult, AuthenticationRequestParameters authenticationRequestParameters,
+                out Dictionary<string, string> allProperties)
+        {
+            allProperties = new Dictionary<string, string>(8, StringComparer.OrdinalIgnoreCase);
+
+            string expiresOn = authResult.ExpiresOn.ToString();
+
+            string extendedExpiresOn = authResult.ExpiresOn.ToString();
+
+            string correlationId = authenticationRequestParameters.CorrelationId.ToString("D");
+
+            if (string.IsNullOrWhiteSpace(correlationId))
+            {
+                _logger.Warning("No correlation ID in response");
+                correlationId = null;
+            }
+
+            MsalTokenResponse msalTokenResponse = new MsalTokenResponse()
+            {
+                AccessToken = authResult.AccessToken,
+                IdToken = authResult.IdToken,
+                CorrelationId = correlationId,
+                Scope = authResult.GrantedScopes,
+                ExpiresIn = DateTimeHelpers.GetDurationFromWindowsTimestamp(expiresOn, _logger),
+                ExtendedExpiresIn = DateTimeHelpers.GetDurationFromWindowsTimestamp(extendedExpiresOn, _logger),
+                ClientInfo = authResult.Account.ClientInfo.ToString(),
+                TokenType = "Bearer",
+                WamAccountId = authResult.Account.Id,
+                TokenSource = TokenSource.Broker
+            };
+
+            return msalTokenResponse;
+
+        }
+#endif
+
+        /// <summary>
+        /// Gets the window handle
+        /// </summary>
+        /// <param name="uiParent"></param>
+        /// <returns></returns>
+        private IntPtr GetParentWindow(CoreUIParent uiParent)
+        {
+#if WINDOWS_APP
+            // On UWP there is no need for a window handle
+            return IntPtr.Zero;
+#endif
+
+#if DESKTOP || NET5_WIN // net core doesn't reference WinForms
+
+            if (uiParent?.OwnerWindow is IWin32Window window)
+            {
+                _logger.Info("[WAM Broker] Owner window specified as IWin32Window.");
+                return window.Handle;
+            }
+#endif
+
+#if DESKTOP || NET5_WIN || NET_CORE
+
+            if (uiParent?.OwnerWindow is IntPtr ptr)
+            {
+                _logger.Info("[WAM Broker] Owner window specified as IntPtr.");
+                return ptr;
+            }
+
+            // other MSALs prefer to default to GetForegroundWindow() but this causes issues
+            // for example if the user quickly switches windows.
+            // GetDesktopWindow will make the default more consistent with the embedded browser
+            _logger.Info("[WAM Broker] Using desktop as a parent window.");
+            IntPtr desktopWindow = WindowsNativeMethods.GetDesktopWindow();
```

Comment:
We should discuss this with msalruntime folks. We need to understand if param needs to be made mandatory.